tests/utils.py

```diff
@@ -44,7 +44,7 @@ def setup_mock_browser(expected_post=None, text=choose_submit_form):
         def text_callback(request, context):
             # Python 2's parse_qsl doesn't like None argument
             query = parse_qsl(request.text) if request.text else ()
-            assert(set(query) == set(expected_post))
+            assert list(query) == list(expected_post)
```

Comment:
Do we need this `list(...)`? If so, this deserves a comment to explain why.

Reply:
The `list` is needed to ensure that `query` and `expected_post` are the same type. I think `query` is always a list, but `expected_post` was sometimes a tuple. I can add a comment, but we could also consider removing the cast for `query`.